app/controllers/messages_controller.rb

```diff
@@ -84,7 +97,17 @@ def preview?
   def permitted_params
     p = model_params.dup
     p.delete(:type)
-    p.permit(permitted_attrs)
+    p = p.permit(permitted_attrs)
+
+    return p unless entry.invoice? && p.dig(:invoice_attributes, :invoice_items_attributes)
```

Comment:
maybe extract in a getter? 

```suggestion
    return p unless invoice_with_attributes?
```